File: simpleClass/scala/src/model/UserInfo.scala
package model

class UserInfo(private var _user: String, private var _password: String) {

  // Get user
  def user = _user

  // Set user
  def user_= (user: String): Unit = _user = user

  // Get password
  def password = this.cryptPassword(_password)

  // Set password
  def password_= (password: String): Unit = _password = password

  // Metodo privado
  private def cryptPassword(password: String): String = "HASH" + password.reverse + "000"
}

File: simpleClass/typescript/src/model/UserInfo.ts
export class UserInfo {
     
    private user: string;
    private password: string;

    constructor(user: string, password: string){
        this.user = user;
        this.password = password;
    }

    public getUser(): string{
        return(this.user);
    }

    public getPassword(): string{
        return(this.cryptPassword(this.password));
    }

    public setUser(user: string) {
        this.user = user;
    }

    public setPassword(password: string) {
        this.password = this.password;
    }

    private cryptPassword(password: string): string {
        return password.split("").reverse().join("");
    }
}